// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.homo_lr";
option java_outer_classname = "HomoLRProto";
option objc_class_prefix = "HL";

package homolr;

service Host {
  rpc GetPubKey (KeyRequest) returns (KeyReply) {}
  rpc AggregateModel (WeightsRequest) returns (WeightsReply) {}
  rpc Validate (Empty) returns (ValidateReply) {}
  rpc Finish (Empty) returns (Empty) {}
  rpc Alive(Empty) returns (Empty) {}
}

message KeyRequest {
  int32 id = 1;
}

message KeyReply {
  bytes key = 1;
}

message WeightsRequest {
  int32 id = 1;
  int32 iter_n = 2;
  bytes weights = 3;
}

message WeightsReply {
  bytes updated_weights = 1;
}

message ValidateReply {
  float acc = 1;
  float loss = 2;
}

message Empty {

}
